### DISCUSSION TITLE
How could Karmada know CPU and memory value ?

### DISCUSSION THREAD

**Discussion by XMingG99** [karmada-io/karmada#2976] [Q&A] [answered]
As we know, when we run command `kubectl get cluster member1 -o yaml --context=karmada-apiserver` ,  we can get a describe of member1. In the describe, which have a field named resourceSummary, in resourceSummary, there have the info of CPU. How could Karmada get this value? By promethus? 
`
resourceSummary:
    allocatable:
      cpu: "4"
      ephemeral-storage: 49805880Ki
      hugepages-1Gi: "0"
      hugepages-2Mi: "0"
      memory: 8147848Ki
      pods: "110"
    allocated:
      cpu: 850m
      memory: 190Mi
    allocating:
      cpu: "0"
      memory: "0"
`

**Comment by RainbowMango** [comment, marked as answer]
The [cluster-status-controller](https://github.com/karmada-io/karmada/blob/c4428d6f224b432a47bfa6aa2bdd6b3a1ccfa275/pkg/controllers/status/cluster_status_controller.go#L211) would sync the resource from clusters.

**Reply by XMingG99** [reply]
ok, thank you

**Reply by xiaxinyu** [reply]
Sorry to interupt,  I also am interesting in this quesiton.
For another questions, cloud karmada perceive the physical resource from hosts that used by clusters?
And How does karmada process where hosts on clusters  haven't  additional physical resouce when client is woking on deploying from karmada?
